### PR TITLE
Refactor get-or-create logic out of __init__()s

### PR DESCRIPTION
**Pull request by convoliution**
This nightmarish find-and-replace job takes the get-or-create logic for `Project`, `Experiment`, and `ExperimentRun` out of `__init__()` and moves them into a new `_get_or_create()`.

This means that these objects can now be instantiated _without extra backend calls or complicated processing_ as long as the caller has a valid entity ID, which can be the case surprisingly often.

### CODE REVIEW

File: client/verta/verta/client.py [hunk 1]
```diff
@@ -1559,7 +1596,7 @@ def _get_records(self, response_msg):
         return response_msg.experiment_runs
 
     def _create_element(self, id_):
-        return ExperimentRun(self._conn, self._conf, _expt_run_id=id_)
```
Comment (convoliution) [inline client/verta/verta/client.py:1534]: This PR has the fortunate consequence of requiring N fewer calls when iterating over `ExperimentRuns`!

File: client/verta/verta/client.py [hunk 2]
```diff
@@ -1199,7 +1197,15 @@ class Project(_ModelDBEntity):
...
-    def __init__(self, conn, conf,
+    def __init__(self, conn, conf, id_):
+        super(Project, self).__init__(conn, conf, _ProjectService, "project", id_)
+
+        self._conn = conn
+
+        self.id = id_
+
+    @staticmethod  # TODO: if PR approved, move this fn down with other static/class methods
+    def _get_or_create(conn, conf,
```
Comment (convoliution) [inline client/verta/verta/client.py]: Here's where the logic is moved out of `Project.__init__()` and into `_get_or_create()`

File: client/verta/verta/client.py [hunk 3]
```diff
@@ -1379,15 +1385,23 @@ class Experiment(_ModelDBEntity):
...
-    def __init__(self, conn, conf,
+    def __init__(self, conn, conf, id_):
+        super(Experiment, self).__init__(conn, conf, _ExperimentService, "experiment", id_)
+
+        self._conn = conn
+
+        self.id = id_
+
+    @staticmethod  # TODO: if PR approved, move this fn down with other static/class methods
+    def _get_or_create(conn, conf,
```
Comment (convoliution) [inline client/verta/verta/client.py]: Here's where the logic is moved out of `Experiment.__init__()` and into `_get_or_create()`

File: client/verta/verta/client.py [hunk 4]
```diff
@@ -1836,7 +1873,15 @@ class ExperimentRun(_ModelDBEntity):
...
-    def __init__(self, conn, conf,
+    def __init__(self, conn, conf, id_):
+        super(ExperimentRun, self).__init__(conn, conf, _ExperimentRunService, "experiment-run", id_)
+
+        self._conn = conn
+
+        self.id = id_
+
+    @staticmethod  # TODO: if PR approved, move this fn down with other static/class methods
+    def _get_or_create(conn, conf,
```
Comment (convoliution) [inline client/verta/verta/client.py]: Here's where the logic is moved out of `ExperimentRun.__init__()` and into `_get_or_create()`

**Review by conradoverta — COMMENTED** [review]
I only see `_get_self_as_msg` being implemented for Experiment. Is this something that you forgot to push or were the others already implemented?

**Comment by convoliution** [comment]
@conradoverta It's already implemented for `ExperimentRun`, and it's not used for `Project`. I think it's a good idea for me to add it to `Project`.

**Review by conradoverta — APPROVED** [review]

**Pull request merged**